tests/deploy_test.go

```diff
@@ -120,6 +121,156 @@ func Test_Deploy_PassingCustomEnvVars_AndQueryString(t *testing.T) {
 	})
 }
 
+func Test_Auto_Scaling(t *testing.T) {
+	labels := map[string]string{
+		"com.openfaas.scale.min": "1",
+		"com.openfaas.scale.max": "2",
+	}
+
+	functionRequest := requests.CreateFunctionRequest{
+		Image:      "functions/alpine:latest",
+		Service:    "test-auto-scaling",
+		Network:    "func_functions",
+		EnvProcess: "env",
+		Labels:     &labels,
+	}
+
+	deployStatus, deployErr := deploy(t, functionRequest)
+	if deployErr != nil {
+		t.Errorf(deployErr.Error())
+		t.Fail()
+		return
+	}
+
+	if deployStatus != http.StatusOK && deployStatus != http.StatusAccepted {
+		t.Logf("got %d, wanted %d or %d", deployStatus, http.StatusOK, http.StatusAccepted)
+		t.Fail()
+	}
+
+	finishHTTP := make(chan bool)
+
+	go func() {
+		end := false
+		for !end {
+			select {
+			case <-finishHTTP:
+				end = true
+			default:
+				invoke(t, functionRequest.Service, emptyQueryString, http.StatusOK)
+			}
+		}
+	}()
+
+	// calling function for 60 seconds to increase # of replicas
+	endSignal := time.After(60 * time.Second)
+	end := false
+
+	for !end {
+		select {
+		case <-endSignal:
+			t.Logf("expected 2 replicas not 1 after calling function for 60 seconds")
+			t.Fail()
+			finishHTTP <- true
+			end = true
+		default:
+			function := get(t, functionRequest.Service)
+
+			if function.Replicas == 2 {
```

Comment:
The use of "2" as a magic number will be hard to maintain. Please extract a variable and use it here and in the inputs.

Suggestion:

```
minReplicas := 1
maxReplicas := 2
```